ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/KeyValueSegmentIdentifier.java

```diff
@@ -25,7 +25,7 @@
 
 public enum KeyValueSegmentIdentifier implements SegmentIdentifier {
   DEFAULT("default".getBytes(StandardCharsets.UTF_8)),
-  BLOCKCHAIN(new byte[] {1}, EnumSet.allOf(DataStorageFormat.class), true, true, false),
+  BLOCKCHAIN(new byte[] {1}, EnumSet.allOf(DataStorageFormat.class), true, true, true),
```

Comment:
As I've found during testing https://github.com/hyperledger/besu/pull/8599
we should be wary of this because it will amplify disk space, even without any pruning occurring.

With this setting, I think you will find the space increases despite the pruning, at least initially.

In my PR I have made it false here and enabled by a flag at runtime when the configuration is available.

Currently I am thinking the user should temporarily enable GC during pruning and then remove again, but I think more testing is required to ensure we don't breach the 2TB limit.